test/powershell/Modules/Microsoft.PowerShell.Utility/WebCmdlets.Tests.ps1

```diff
@@ -1174,32 +1151,54 @@ Describe "Invoke-WebRequest tests" -Tags "Feature" {
-            ConvertFrom-Json
+    Context "HTTPS Tests" {
+        It "Validate Invoke-WebRequest -SkipCertificateCheck" {
+            # validate that exception is thrown for URI with expired certificate
+            $Uri = Get-WebListenerUrl -Https
+            $command = "Invoke-WebRequest -Uri '$Uri'"
+            $result = ExecuteWebCommand -command $command
+            $result.Error.FullyQualifiedErrorId | Should Be "WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
+            
+            # validate that no exception is thrown for URI with expired certificate when using -SkipCertificateCheck option
+            $Uri = Get-WebListenerUrl -Https
+            $command = "Invoke-WebRequest -Uri '$Uri' -SkipCertificateCheck"
+            $result = ExecuteWebCommand -command $command
+            $result.Error | Should BeNullOrEmpty
+        }
         
-        $result.Status  | Should Be 'FAILED'
-    }
-    
-    # Test skipped on macOS pending support for Client Certificate Authentication
-    # https://github.com/PowerShell/PowerShell/issues/4650
-    It "Verifies Invoke-WebRequest Certificate Authentication Successful with -Certificate" -skip:$IsOSX {
-        $uri = Get-WebListenerUrl -Https -Test 'Cert'
-        $certificate = Get-WebListenerClientCertificate
-        $result = Invoke-WebRequest -Uri $uri -Certificate $certificate -SkipCertificateCheck | 
-            Select-Object -ExpandProperty Content |
-            ConvertFrom-Json
+        It "Validate Invoke-WebRequest returns native HTTPS error message in exception" {
+            $uri = Get-WebListenerUrl -Https
+            $command = "Invoke-WebRequest -Uri '$uri'"
+            $result = ExecuteWebCommand -command $command
+            
+            # need to check against inner exception since Linux and Windows uses different HTTP client libraries so errors aren't the same
+            $result.Error.ErrorDetails.Message | Should Match $result.Error.Exception.InnerException.Message
+            $result.Error.FullyQualifiedErrorId | Should Be "WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeWebRequestCommand"
+        }
         
-        $result.Status  | Should Be 'OK'
-        $result.Thumbprint | Should Be $certificate.Thumbprint
+        It "Verifies Invoke-WebRequest Certificate Authentication Fails without -Certificate"  {
+            $uri = Get-WebListenerUrl -Https -Test 'Cert'
+            $result = Invoke-WebRequest -Uri $uri -SkipCertificateCheck | 
+                Select-Object -ExpandProperty Content |
+                ConvertFrom-Json
+            
+            $result.Status | Should Be 'FAILED'
+        }
+        
+        # Test skipped on macOS pending support for Client Certificate Authentication
+        # https://github.com/PowerShell/PowerShell/issues/4650
+        It "Verifies Invoke-WebRequest Certificate Authentication Successful with -Certificate" -skip:$IsOSX {
```

Comment:
if this is eventually going to be run on mac, could you make this `-pending:$IsOSX`? we track pending tests more closely than skipped tests.

Reply:
I will switch them to pending. I was asked to "skip" them but perhaps they didn't mean literally to use `-skip`

Reply:
Ah, my bad.